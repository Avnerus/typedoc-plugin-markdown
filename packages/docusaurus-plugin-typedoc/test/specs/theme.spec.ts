import * as Handlebars from 'handlebars';
import * as path from 'path';
import { TestApp } from '../../../typedoc-plugin-markdown/test/test-app';
import DocusaurusTheme from '../../src/theme/theme';

describe(`Theme:`, () => {
  let testApp: TestApp;
  let reflectionTemplate: Handlebars.TemplateDelegate;

  beforeAll(() => {
    testApp = new TestApp(['theme.ts']);
    testApp.bootstrap({
      theme: path.resolve(__dirname, '..', '..', 'dist', 'theme'),
    });
    TestApp.stubPartials(['index', 'comment', 'member.signature', 'members']);
    TestApp.stubHelpers(['breadcrumbs', 'hierarchy']);
    reflectionTemplate = TestApp.getTemplate('reflection');
  });

  describe(`(mdx)`, () => {
    test(`should escape angle brackets and quotes for full mdx support`, () => {
      const content = 'Escape \\<angle-brackets> and \\"quotes\\"';
      expect(DocusaurusTheme.formatContents(content)).toMatchSnapshot();
    });
  });

  describe(`(header)`, () => {
    test(`should compile with module without subtitle`, () => {
      expect(
        TestApp.compileTemplate(reflectionTemplate, {
          model: testApp.project.children[0],
          project: testApp.project,
        }),
      ).toMatchSnapshot();
    });

    test(`should compile with module with subtitle`, () => {
      expect(
        TestApp.compileTemplate(reflectionTemplate, {
          model: testApp.findReflection('InterfaceItemA').children[0],
          project: testApp.project,
        }),
      ).toMatchSnapshot();
    });
  });
});
