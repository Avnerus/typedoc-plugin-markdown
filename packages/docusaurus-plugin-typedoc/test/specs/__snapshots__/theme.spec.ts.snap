// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Theme: (header) should compile with module with subtitle 1`] = `
"
"
`;

exports[`Theme: (header) should compile with module without subtitle 1`] = `
"
"
`;

exports[`Theme: (mdx) should escape angle brackets and quotes for full mdx support 1`] = `"Escape &#60;angle-brackets> and &#34;quotes&#34;"`;
